src/liboslexec/shadingsys.cpp

```diff
@@ -265,9 +265,26 @@ ShadingSystem::execute (ShadingContext &ctx, ShaderGroup &sas,
 
 
 const void*
-ShadingSystem::get_symbol (ShadingContext &ctx, ustring name, TypeDesc &type)
+ShadingSystem::get_symbol (ShadingContext &ctx, ustring layername,
```

Comment:
Revision: it definitely returns NULL if the layer was *eliminated* during optimization. But I think it returns the valid memory location (whose contents will be uninitialized) if the layer existed but was never run (or hasn't run *yet*).